code/game/machinery/firealarm.dm

```diff
@@ -326,3 +326,121 @@
 	if (!party_overlay)
 		party_overlay = iconstate2appearance('icons/turf/areas.dmi', "party")
 	A.add_overlay(party_overlay)
+
+/obj/machinery/sprinkler
+	name = "sprinkler"
+	desc = "An automated sprinkler capable of detecting fire and spraying coolant."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "sprinkler0"
+	layer = PROJECTILE_HIT_THRESHHOLD_LAYER
+	plane = FLOOR_PLANE
+	max_integrity = 100
+	armor = list("melee" = 50, "bullet" = 20, "laser" = 20, "energy" = 20, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 80, "acid" = 30)
+	var/refill_loaded = FALSE
+	req_access = list(ACCESS_ENGINE)
+	var/cooldown = 0
+
+/obj/machinery/sprinkler/Initialize()
+	. = ..()
+	internal = new /obj/item/reagent_containers/glass/beaker(src)
+
+/obj/machinery/sprinkler/interact(mob/user)
+	if(!allowed(user))
+		to_chat(user, "<span class='danger'>Access denied.</span>")
+		return
+	dispense()
+
+/obj/machinery/sprinkler/emag_act(mob/user)
+	. = ..()
+	if(obj_flags & EMAGGED)
+		return
+	playsound(src, "sparks", 75, 1)
+	obj_flags |= EMAGGED
+	internal.reagents.remove_all(50)
+	internal.reagents.add(pick(/datum/reagent/clf3,
+							   /datum/reagent/drug/space_drugs,
+							   /datum/reagent/lube,
+							   /datum/reagent/fuel/unholywater,

```

Comment:
Unholy water outside of cult is big oof. Go back to hippie with that.

Should also probably be internal.reagents.remove_all(internal.reagents.total_volume)